src/jwt.rs

```diff
@@ -175,6 +175,9 @@ where
 
     pub fn decode(input: &str) -> Result<Claims<T>> {
         let segments: Vec<&str> = input.split('.').collect();
+        if segments.len() != 3 {
+            return Err(errors::new(errors::ErrorKind::Token(input.into())));
```

Comment:
Probably don't want to dump the entire token as the error message, maybe just "improperly formed token" or something similar.